src/mongo.c

```diff
@@ -1218,6 +1238,10 @@ static int mongo_cursor_op_query( mongo_cursor *cursor ) {
 
     if( cursor->reply->fields.num == 1 ) {
         bson_init_finished_data( &temp, &cursor->reply->objs, 0 );
+
+        if( mongo_check_bson_overflow( temp.dataSize, cursor ) == MONGO_ERROR )
```

Comment:
@arsaha-elastica Please note that we cannot switch this function towards end when we are returning MONGO_OK since that is failing gridfs UT case. Also we cannot add this sanity check with `mongo_cursor_get_more` with additional check for NULL cursor->data since it would have no effect as curosr->data is set to NULL just before returning MONGO_OK
```
1311 
1312         cursor->current.data = NULL;
1313         cursor->seen += cursor->reply->fields.num;
1314 
```